src/Server/TCPHandler.cpp

```diff
@@ -402,14 +402,11 @@ void TCPHandler::runImpl()
                     {
                         auto callback = [this]()
                         {
-                            {
-                                std::lock_guard task_callback_lock(task_callback_mutex);
+                            std::lock_guard lock1(task_callback_mutex);
+                            std::lock_guard lock2(fatal_error_mutex);
```

Comment:
Maybe use https://en.cppreference.com/w/cpp/thread/scoped_lock so deadlock avoidance is used?